src/vs/workbench/contrib/notebook/browser/view/renderers/backLayerWebView.ts

```diff
@@ -707,36 +707,19 @@ export class BackLayerWebView<T extends ICommonCellInfo> extends Disposable {
 				</script>
 				${coreDependencies}
 				<div id='container' class="widgetarea" style="position: absolute;width:100%;top: 0px"></div>
-				<script type="module">${preloadsScriptStr(this.options, markupRenderer)}</script>
+				<script type="module">${preloadsScriptStr(this.options, renderersData)}</script>
 			</body>
 		</html>`;
 	}
 
-	private getMarkdownRenderer(): WebviewPreloadRenderer[] {
-		const markdownMimeType = 'text/markdown';
-		const allRenderers = this.notebookService.getRenderers()
-			.filter(renderer => renderer.matchesWithoutKernel(markdownMimeType) !== NotebookRendererMatch.Never);
-
-		const topLevelMarkdownRenderers = allRenderers
-			.filter(renderer => renderer.dependencies.length === 0);
-
-		const subRenderers = new Map<string, Array<{ entrypoint: string }>>();
-		for (const renderer of allRenderers) {
-			for (const dep of renderer.dependencies) {
-				if (!subRenderers.has(dep)) {
-					subRenderers.set(dep, []);
-				}
-				const entryPoint = this.asWebviewUri(renderer.entrypoint, renderer.extensionLocation);
-				subRenderers.get(dep)!.push({ entrypoint: entryPoint.toString(true) });
-			}
-		}
-
-		return topLevelMarkdownRenderers.map((renderer): WebviewPreloadRenderer => {
-			const src = this.asWebviewUri(renderer.entrypoint, renderer.extensionLocation);
+	private getRendererData(): RendererMetadata[] {
+		return this.notebookService.getRenderers().map((renderer): RendererMetadata => {
```

Comment:
I wonder if we can build this in such a way that manual handling of markdown renderers is not needed. For example, if all `extends` (once the 2nd PR is ready) installed are automatically loaded whenever the renderer they extend is loaded.

Reply:
Oh, I see this was added so `getRenderers()` returns every installed renderer, ok. May need to eventually handling dynamically installing renderer extensions later...